File: http/src/main/java/com/xinzy/http/RequestCallback.java
package com.xinzy.http;

/**
 * Created by Xinzy on 2017/6/12.
 *
 */
public interface RequestCallback<T> {

    void onSuccess(T t);
    void onFailure(SmartHttpException e);

    class DefaultRequestCallback<T> implements RequestCallback<T> {

        @Override
        public void onSuccess(T t) { }

        @Override
        public void onFailure(SmartHttpException e) { }
    }
}

File: httpkotlin/src/main/java/com/xinzy/http/kotlin/RequestCallback.kt
package com.xinzy.http.kotlin

/**
 * Created by Xinzy on 2018/10/12.
 */
interface RequestCallback<T> {
    fun onSuccess(t: T?)
    fun onFailure(e: SmartHttpException)
}

class DefaultRequestCallback<T> : RequestCallback<T> {
    override fun onSuccess(t: T?) {}
    override fun onFailure(e: SmartHttpException) {}
}